### ISSUE TITLE
bug: scrollBottom() removes focus on input or textareas

### ISSUE THREAD

**Issue by clawish** [driftyco/ionic#2904]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>When $ionicScrollDelegate.scrollBottom() and .scrollTop() are fired when an input or textarea field is focused, the focus is lost! This behavior is new to beta 14.
See this codepen for the issue: http://codepen.io/tsurrel/pen/myRmbe
Just type in something in the input field and see how focus is lost, because .scrollBottom() is called. Then replace the beta 14 by beta 13 and see how focus is kept (this should be the intended behavior).
Thank you for fixing.</span>

<span is-issue-template></span>

**Comment by ckken** [comment]
just rebult the function remove ..blurall andfix it

**Comment by clawish** [comment]
Removing blurall works.
So this is actually intended... Sorry for compaining then.

**Comment by cquartier** [comment]
@clawish this is intended behavior? I feel like input blurring should not be a concern of the scroll delegate. In my case, I do **NOT** want inputs to blur on scrolling.

**Comment by pulakb** [comment]
@clawish can you please share solution using a codepen?

**Comment by clawish** [comment]
in the `ionic.bundle.js` change the lines

    self.scrollTop = function(shouldAnimate) {
        ionic.DomUtil.blurAll();
        self.resize().then(function() {
            scrollView.scrollTo(0, 0, !!shouldAnimate);
        });
    };

    self.scrollBottom = function(shouldAnimate) {
        ionic.DomUtil.blurAll();
        self.resize().then(function() {
            var max = scrollView.getScrollMax();
            scrollView.scrollTo(max.left, max.top, !!shouldAnimate);
        });
    };

to

    self.scrollTop = function(shouldAnimate) {
        //ionic.DomUtil.blurAll();
        self.resize().then(function() {
            scrollView.scrollTo(0, 0, !!shouldAnimate);
        });
    };

    self.scrollBottom = function(shouldAnimate) {
        //ionic.DomUtil.blurAll();
        self.resize().then(function() {
            var max = scrollView.getScrollMax();
            scrollView.scrollTo(max.left, max.top, !!shouldAnimate);
        });
    };

**Comment by panurge-ws** [comment]
I suggest to add a variable to avoid the blurring, instead of commenting out:
```javascript
   self.scrollTop = function(shouldAnimate, avoidBlur) {
    if (avoidBlur !== true){
      ionic.DomUtil.blurAll();
    }
    self.resize().then(function() {
      scrollView.scrollTo(0, 0, !!shouldAnimate);
    });
  };
```

**Comment by osahner** [comment]
@panurge-ws nice solution! 
Would you be so kind and create a pull request?
Until now, the "bug" is still in the code.

**Comment by osahner** [comment]
Can be closed: 0145dc3
